core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -998,11 +998,38 @@ public static void udpateUserRoleMappingInBatchMode(Connection dbConnection, Str
             }
             dbConnection.commit();
         } catch (SQLException e) {
-            String errorMessage = "Using sql : " + sqlStmt + " " + e.getMessage();
-            if (log.isDebugEnabled()) {
-                log.debug(errorMessage, e);
+            boolean isUniqueKeyViolationException = false;
+            for (Throwable subSQLException : e) {
+                if (subSQLException instanceof SQLIntegrityConstraintViolationException ||
+                        StringUtils.containsIgnoreCase(e.getMessage(),
+                                "um_user_role_um_user_id_um_role_id_um_tenant_id_key")) {
```

Comment:
Is this a necessary condition ? And can we ensure that this part(`um_user_role_um_user_id_um_role_id_um_tenant_id_key `) is included in the error messages from all the supported database types ?

Reply:
This is for postgres db only. others will catch with SQLIntegrityConstraintViolationException